src/server/tran_server.cpp

```diff
@@ -454,6 +454,21 @@ tran_server::uses_remote_storage () const
   return false;
 }
 
+bool tran_server::get_main_connection_info (std::string &host_out, int32_t &port_out)
+{
+  auto slock = std::shared_lock<std::shared_mutex> { m_main_conn_mtx };
+  if (m_main_conn == nullptr)
+    {
+      return false;
+    }
+
+  auto &node = m_main_conn->get_node ();
```

Comment:
Nit.
Although maybe superfluous as `auto` == `const cubcomm::node` already. Not sure. Some `decltype` debugging should tell you if superfluous or not.
```suggestion
  const auto &node = m_main_conn->get_node ();
```

Reply:
If it's not reference, the const would be cut off, but it's read-only as it is. I was not sure, but I've tested it.